bson/_cbsonmodule.c

```diff
@@ -510,8 +536,24 @@ static int write_element_to_buffer(PyObject* self, buffer_t buffer,
     int result;
     if(Py_EnterRecursiveCall(" while encoding an object to BSON "))
         return 0;
+
+    /* Transform types that have a registered converter.
+     * A new reference is created upon transformation. */
+    PyObject* value_type = PyObject_Type(value);
```

Comment:
These declarations have to go above the if block above or this won't compile in VC++.